core/src/main/java/org/apache/carbondata/core/datastore/block/SegmentPropertiesAndSchemaHolder.java

```diff
@@ -130,16 +137,32 @@ public int addSegmentProperties(CarbonTable carbonTable,
         }
       }
     } else {
-      synchronized (getOrCreateTableLock(carbonTable.getAbsoluteTableIdentifier())) {
+      synchronized (getOrCreateTableLock(
+          segmentPropertiesWrapper.carbonTable.getAbsoluteTableIdentifier())) {
+        updateDeleteMap(segmentIdSetAndIndexWrapper.getSegmentPropertiesIndex(), -1);
```

Comment:
It is better we can it only it is already present in delete map

Reply:
okay